app/src/main/java/com/example/habitsmasher/ui/history/HabitEventViewFragment.java

```diff
@@ -51,4 +81,75 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
 
         return view;
     }
+
+    @Override
+    public void onSaveInstanceState(Bundle outState) {
+        super.onSaveInstanceState(outState);
+
+        Bundle mapViewBundle = outState.getBundle(MAPVIEW_BUNDLE_KEY);
+        if (mapViewBundle == null) {
+            mapViewBundle = new Bundle();
+            outState.putBundle(MAPVIEW_BUNDLE_KEY, mapViewBundle);
+        }
+
+        _mapView.onSaveInstanceState(mapViewBundle);
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        if (!_habitEvent.getLocation().equals("")) {
+            _mapView.onResume();
+        }
+    }
```

Comment:
This ensures the program attempts to treat _mapView as a fragment when a habit event doesn't have a location